element_animal/surgery.py

```diff
@@ -0,0 +1,169 @@
+
+schema = dj.Schema()
+
+_linking_module = None
+
+
+def activate(
+    surgery_schema_name: str,
+    subject_schema_name: str = None,
+    *,
+    create_schema: bool = True,
+    create_tables: bool = True,
+    linking_module: bool = True
+):
+    """Activate this schema.
+
+    Args:
+        schema_name (str): schema name on the database server to activate the
+                        `subject` element
+        create_schema (bool): when True (default), create schema in the
+                            database if it does not yet exist.
+        create_tables (bool): when True (default), create tables in the
+                            database if they do not yet exist.
+        linking_module (bool): a module name or a module containing the
+        required dependencies to activate the `subject` element:
+
+    Dependencies:
+    Upstream tables:
+        User: the who conducted a particular surgery/implantation
+    """
+
+    if isinstance(linking_module, str):
+        linking_module = importlib.import_module(linking_module)
+    assert inspect.ismodule(
+        linking_module
+    ), "The argument 'linking_module' must be a module's name or a module"
+
+    global _linking_module
+    _linking_module = linking_module
+
+    subject.activate(
+        subject_schema_name,
+        create_schema=create_schema,
+        create_tables=create_tables,
+        linking_module=linking_module,
+    )
+    schema.activate(
+        surgery_schema_name,
+        create_schema=create_schema,
+        create_tables=create_tables,
+        add_objects=linking_module.__dict__,
+    )
+
+
+@schema
+class CoordinateReference(dj.Lookup):
+    """Coordinate reference system
+
+    Attributes:
+        reference ( varchar(60) ): Reference system (e.g., bregma, lambda, etc.)
+    """
+
+    definition = """
+    reference   : varchar(60)
+    """
+    contents = zip(
+        ["bregma", "lambda", "dura", "skull_surface", "sagittal_suture", "sinus"]
+    )
+
+
+@schema
+class BrainRegion(dj.Manual):
+    """Brain region of a given surgery
+
+    Attributes:
+        region_acronym ( varchar(32) ) : Brain region shorthand
+        region_name ( varchar(128) ) : Brain region full name
+    """
+
+    definition = """
+    region_acronym : varchar(32)   # Brain region shorthand
+    ---
+    region_name    : varchar(128)  # Brain region full name
+    """
+
+
+@schema
+class Hemisphere(dj.Lookup):
+    """Brain region hemisphere
+
+    Attributes:
+        hemisphere ( varchar(8) ): Brain region hemisphere (e.g., left, right, middle)
+    """
+
+    definition = """
+    hemisphere: varchar(8) # Brain region hemisphere
+    """
+
+    contents = zip(["left", "right", "middle"])
+
+
+@schema
+class ImplantationType(dj.Lookup):
+    """Type of implantation
+
+    Attributes:
+        implant_type ( varchar(16) ): Short name for type of implanted device
+        implant_description ( varchar(32) ): Full description for implanted device
+    """
```

Comment:
Looks like this table stores implantation types, not the actual implanted device. I think most commonly implanted device in neuroscience can be things like electrode, cannula, probe, fiber, microdrive, etc.  `"opto"` or `"ephys"` can be confusing as there can be multiple devices of that type and some support both (e.g., "optrode").

Reply:
How would you suggest changing this table? Change to ImplantedDevice with a short name and full description?